testdata/inline-request.txt

```diff
@@ -0,0 +1,62 @@
+# Tests a request with an inline request body
+
+openapi-httprequest api.yaml
+cmp api-params.go expect-api-params.go
+
+-- api.yaml --
+openapi: 3.0.0
+info:
+  description: test service description
+  title: test service
+  version: 1.0.0
+paths:
+  /test:
+    post:
+      description: Post test data (description)
+      operationId: PostTestData
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                test:
+                  type: string
+                  description: A test string
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PostResponse'
+      summary: Get test data
+components:
+  schemas:
+    PostResponse:
+      properties:
+        body:
+          description: post test data response body
+          type: string
+      type: object
+
+-- expect-api-params.go --
+// Code generated by openapi-httprequest. DO NOT EDIT.
+
+package params
+
+import (
+	httprequest "gopkg.in/httprequest.v1"
+)
+
+type APIHandler interface {
+	PostTestData(httprequest.Params, *PostTestDataRequest) (*PostResponse, error)
+}
+
+type PostResponse struct {
+	Body string `json:"body"`
+}
+
+type PostTestDataRequest struct {
+	httprequest.Route `httprequest:"POST /test"`
+	Test              string `httprequest:"test,body"`
```

Comment:
This doesn't look quite right to me. AIUI, the YAML encodes a JSON body that is an object containing a string field named "test", but this represents a JSON body that is itself just a string.

Reply:
Oh really, I did wonder this come to think of it.
What is the correct way of showing this? Would it simply be:
```
type PostTestDataRequest struct {
	httprequest.Route `httprequest:"POST /test"`
	Test              string `json:"test"`
}
```

Reply:
If you want a JSON body that's just a string, then:

```
type PostTestDataRequest struct {
	httprequest.Route `httprequest:"POST /test"`
	Test              string `httprequest:",body"`
}
```

If you want a JSON object contains a string field "test" then either:

```
type PostTestDataRequest struct {
	httprequest.Route `httprequest:"POST /test"`
	Body              PostTestDataRequestBody  `httprequest:",body"`
}

type PostTestDataRequestBody struct {
    Test string `json:"test"`
}
```

or:
```
type PostTestDataRequest struct {
	httprequest.Route `httprequest:"POST /test"`
	Body              struct {
		Test string `json:"test"`
	} `httprequest:",body"`
}
```

or even:

```
type PostTestDataRequest struct {
	httprequest.Route `httprequest:"POST /test"`
	PostTestDataRequestBody  `httprequest:",body"`
}

type PostTestDataRequestBody struct {
    Test string `json:"test"`
}
```